packages/core-v2-sdk/src/models/Token.ts

```diff
@@ -0,0 +1,57 @@
+import { ElementClient } from "src/client";
+import { TokenContractDataSource } from "src/datasources/Token/TokenContractDataSource";
+import { TokenDataSource } from "src/datasources/Token/TokenDataSource";
+import { CoinGeckoAPIDataSource } from "src/datasources/TokenAPI/CoinGeckoAPIDataSource";
+
+export class Token {
+  address: string;
+  client: ElementClient;
+  dataSource: TokenDataSource;
+
+  constructor(
+    address: string,
+    client: ElementClient,
+    dataSource?: TokenDataSource,
+  ) {
+    this.address = address;
+    this.client = client;
+    if (dataSource) {
+      this.dataSource = dataSource;
```

Comment:
we don't want to cache a custom datasource here?

Reply:
None of the models register custom data sources. I don't know if they should assume that was your intention. What if I want 2 instances of a model that both use different datasources with the same unique id?

I think if the user is going to pass a custom datasource to a model, then they take on the responsibility of registering it with the context.